happy-birthday-to-you.js

```diff
@@ -0,0 +1,153 @@
+/* 
+
+------ STEPS ------
+1.  Ensure that name is in correct format (only strings, also account for 
+	special characters).
+
+2. Once correct name format is inputted, compare it to object key/value pairs.
+
+2a. Play a birthday song if dates match. otherwise, play the Rick Roll 
+
+*/
+
+const birthdayObj = {
+
+	'Abdullah Sayeem': 513,
+	'Andre Chow': 103,
+	'Gurdeep Singh': 327,
+	'Jimmy Feng': 711,
+	'John Chen': 1202,
+	'Kevin Chen': 627,
+	'Kevin Cheung':	521,
+	'Mohammed Rahman': 305,
+	'Robert Tan': 523,
+	'Roland Zhou': 903,
+	'Vainius Glinskis': 805,
+	'Zarif Rahat': 1107,
+
+};
+
+function validateInputName() {
+	let text = nameInput.value.toLowerCase();
+	// any combination of letters separated by a whitespace
+	let nameFormat = /^[a-zA-Z]+\s+[a-zA-Z]+$/; 
+
+	return nameFormat.test(text);
+}
+
+function confirmOnList() {
+	let text = nameInput.value.toLowerCase();
+
+	for (let name in birthdayObj) {
+		name = name.toLowerCase();
+
+		if (text === name) {
+			return true;
+		}
+	}
+
+	return false;
+}
```

Comment:
So an alternative would be to leverage the fact that objects can lookup keys really quickly.

To confirm, what you could do is capitalize the name input, and then just see whether the object can look that name up.

For example, 

if I typed `john chen`, you can capitalize that to be `John Chen`, then just do 

`return !!birthdayObj[nameInputValue]`

Let me know if that makes sense.